app/forms/hyrax/forms/resource_batch_edit_form.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+module Hyrax
+  module Forms
+    class ResourceBatchEditForm < Hyrax::Forms::ResourceForm
+      include Hyrax::FormFields(:basic_metadata)
+
+      self.required_fields = []
+      self.model_class = Hyrax.primary_work_type
```

Comment:
does this form only support the primary work type?

Reply:
No. It'll work for any models, even with a batch  that mixes with different objects types. The model_class here is only served for retrieving a consistent `param_key` for different object types.